rust/cw-contracts/nft-launchpad/src/contract.rs

```diff
@@ -0,0 +1,348 @@
+                minter: sender,
+                launchpad_contract: "launchpad_contract".to_string(),
+                mint_info: Tr721MintInfo {
+                    metadatas_merkle_root: collection.metadatas_merkle_root.unwrap(),
+                    tokens_count: collection.tokens_count,
+                    royalty_address: collection.royalty_address,
+                    royalty_percentage: collection.royalty_percentage,
+                },
+                mint_periods: collection.mint_periods,
+            })?,
+            funds: vec![],
+            label: format!(
+                "TR721 codeId:{} collectionId:{} symbol:{}",
+                nft_code_id.unwrap(),
+                collection_id,
+                collection.symbol
+            ),
+        };
+
+        // Update instantiating collection id
+        self.instantiating_collection_id
+            .save(ctx.deps.storage, &collection_id)?;
+
+        let submessage = SubMsg::reply_on_success(instantiate_msg, INSTANTIATE_REPLY_ID);
+
+        Ok(Response::new()
+            .add_submessage(submessage)
+            .add_attribute("action", "collection_deployed")
+            .add_attribute("collection_id", collection_id))
+    }
+
+    #[msg(query)]
+    pub fn get_collection_by_id(
+        &self,
+        ctx: QueryCtx,
+        collection_id: String,
+    ) -> StdResult<Collection> {
+        let collection = self.collections.load(ctx.deps.storage, collection_id)?;
+        Ok(collection)
+    }
+
+    #[msg(query)]
+    pub fn get_config(&self, ctx: QueryCtx) -> StdResult<Config> {
+        let config = self.config.load(ctx.deps.storage)?;
+        Ok(config)
+    }
+
+    #[msg(reply)]
+    pub fn handle_reply(&self, ctx: ReplyCtx, msg: Reply) -> Result<Response, ContractError> {
+        let msg_id = msg.id;
+        let storage = ctx.deps.storage;
+
+        if msg_id == INSTANTIATE_REPLY_ID {
+            let resp = parse_reply_instantiate_data(msg).unwrap();
+            let deployed_addr = resp.contract_address;
+
+            // Get instantiating collection id
+            let collection_id = self.instantiating_collection_id.load(storage)?;
+
+            // Update collection states
+            let mut collection = self.collections.load(storage, collection_id.to_owned())?;
+            collection.deployed_address = Some(deployed_addr.clone());
+            self.collections
+                .save(storage, collection_id.to_owned(), &collection)?;
+
+            return Ok(Response::new()
+                .add_attribute("action", "collection_instantiated")
+                .add_attribute("collection_id", collection_id)
+                .add_attribute("collection_addr", deployed_addr));
+        }
+
+        Err(ContractError::UnknownReply { reply_id: msg_id })
+    }
+}
+
+// Types definitions ------------------------------------------------------
+#[cw_serde]
+pub struct Config {
+    pub name: String,
+    pub nft_code_id: Option<u64>,
```

Comment:
why make it an `Option`?
```suggestion
    pub nft_code_id: u64,
```

Reply:
I modified the Config

Reply:
when do we want only parts of the functions working?

Reply:
I don't understand.
We don't need tr721 to send our CollectionProject data

Reply:
yes okay, but do we really want to allow to instantiate the launchpad with no underlying cw721 implem or no admin?

Reply:
For me, if we don't need it to execute things, it should be not mandatory (dummy style)
But it's not a "smartcontract vision".

I can counter my proper argument with this: https://github.com/TERITORI/teritori-dapp/blob/feat-nft-launchpad-contract/rust/cw-contracts/cw721-membership/src/contract.rs#L90
All `Config` fields are mandatory, but not necessary used in all contract's functions.

I have no use case, I'll make them mandatory ^^

Reply:
https://github.com/TERITORI/teritori-dapp/pull/1315/commits/963b8391cd35228550f1309381df292ed4b1211c